lib/web_ui/lib/src/engine/window.dart

```diff
@@ -478,14 +480,30 @@ class EngineWindow extends ui.Window {
   /// Engine code should use this method instead of the callback directly.
   /// Otherwise zones won't work properly.
   void invokeOnPlatformMessage(
-      String name, ByteData? data, ui.PlatformMessageResponseCallback callback) {
-    _invoke3<String, ByteData?, ui.PlatformMessageResponseCallback>(
-      _onPlatformMessage,
-      _onPlatformMessageZone,
-      name,
-      data,
-      callback,
-    );
+    String name,
+    ByteData? data,
+    ui.PlatformMessageResponseCallback callback,
+  ) {
+    if (name == ui.ChannelBuffers.kControlChannelName) {
+      // TODO(ianh): move this logic into ChannelBuffers once we remove onPlatformMessage
+      try {
+        channelBuffers.handleMessage(data!);
+      } catch (ex) {
+        print('Message to "$name" caused exception $ex');
```

Comment:
This should probably log to https://api.dart.dev/stable/2.10.1/dart-html/Console/error.html, and should also include the stack trace.

Reply:
Yeah I was wondering about this. Maybe we shouldn't even catch the exception here, and let the usual uncaught exception logic do its thing? I've removed the catch entirely for now. cc @gaaclarke who added this catch originally.